File: app/src/main/java/me/ycdev/android/devtools/sampler/cpu/ProcCpuStat.java
package me.ycdev.android.devtools.sampler.cpu;

public class ProcCpuStat {
    public int pid;
    public long utime; // user mode
    public long stime; // kernel mode

    public ProcCpuStat(int pid) {
        this.pid = pid;
    }

    public long getTotal() {
        return utime + stime;
    }

    public static ProcCpuStat computeUsage(ProcCpuStat oldStat, ProcCpuStat newStat) {
        if (oldStat.pid != newStat.pid) {
            return null;
        }
        ProcCpuStat usage = new ProcCpuStat(newStat.pid);
        usage.utime = newStat.utime - oldStat.utime;
        usage.stime = newStat.stime - oldStat.stime;
        return usage;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("ProcCpuStat[pid=").append(pid);
        sb.append(", utime=").append(utime);
        sb.append(", stime=").append(stime);
        sb.append("]");
        return sb.toString();
    }
}

File: app/src/main/java/me/ycdev/android/devtools/sampler/cpu/ProcCpuStat.kt
package me.ycdev.android.devtools.sampler.cpu

data class ProcCpuStat(var pid: Int) {
    var utime: Long = 0 // user mode
    var stime: Long = 0 // kernel mode

    val total: Long get() = utime + stime

    companion object {
        fun computeUsage(oldStat: ProcCpuStat, newStat: ProcCpuStat?): ProcCpuStat? {
            if (oldStat.pid != newStat!!.pid) {
                return null
            }
            val usage = ProcCpuStat(newStat.pid)
            usage.utime = newStat.utime - oldStat.utime
            usage.stime = newStat.stime - oldStat.stime
            return usage
        }
    }
}
